streams/src/test/java/org/apache/kafka/streams/kstream/internals/SuppressScenarioTest.java

```diff
@@ -529,6 +532,74 @@ public void shouldSupportFinalResultsForSessionWindows() {
         }
     }
 
+    @Test
+    public void shouldApplyNameToFinalSuppressionNode() {
+        final StreamsBuilder anonymousNodeBuilder = new StreamsBuilder();
+        anonymousNodeBuilder
+            .stream("input", Consumed.with(STRING_SERDE, STRING_SERDE))
+            .groupBy((String k, String v) -> k, Serialized.with(STRING_SERDE, STRING_SERDE))
+            .windowedBy(SessionWindows.with(5L).grace(5L))
+            .count(Materialized.<String, Long, SessionStore<Bytes, byte[]>>as("counts").withCachingDisabled())
+            .suppress(untilWindowCloses(unbounded()))
+            .toStream()
+            .map((final Windowed<String> k, final Long v) -> new KeyValue<>(k.toString(), v))
+            .to("output-suppressed", Produced.with(STRING_SERDE, Serdes.Long()));
+        final String anonymousNodeTopology = anonymousNodeBuilder.build().describe().toString();
+
+        // without the name, the suppression node increments the topology index
+        assertThat(anonymousNodeTopology, containsString("0000000009"));
+        assertThat(anonymousNodeTopology, not(containsString("0000000010")));
+
+        final StreamsBuilder namedNodeBuilder = new StreamsBuilder();
+        namedNodeBuilder
+            .stream("input", Consumed.with(STRING_SERDE, STRING_SERDE))
+            .groupBy((String k, String v) -> k, Serialized.with(STRING_SERDE, STRING_SERDE))
+            .windowedBy(SessionWindows.with(5L).grace(5L))
+            .count(Materialized.<String, Long, SessionStore<Bytes, byte[]>>as("counts").withCachingDisabled())
+            .suppress(untilWindowCloses(unbounded()).withName("myname"))
+            .toStream()
+            .map((final Windowed<String> k, final Long v) -> new KeyValue<>(k.toString(), v))
+            .to("output-suppressed", Produced.with(STRING_SERDE, Serdes.Long()));
+        final String namedNodeTopology = namedNodeBuilder.build().describe().toString();
+
+        // without the name, the suppression node does not increment the topology index
+        assertThat(namedNodeTopology, containsString("myname"));
```

Comment:
I'm thinking we should also compare the expected topology against the one generated here.  This way any other changes that could affect the topology will surface.  We've done this in a lot of our unit tests where we build topologies.

Reply:
Ok, yeah, that's a good idea.